design/Robust/empirical.Rmd

```diff
@@ -0,0 +1,124 @@
+---
+title: "Design for empirical covariance"
+author: "Liming Li"
+output: html_document
+editor_options:
+  chunk_output_type: console
+---
+
+# Implementations in SAS
+
+According to SAS documentation, the robust sandwich estimator is implemented using "empirical" option.
+With this option enabled, the method "Satterthwaite" and "Kenward-Roger" are not available.
+A simple example in SAS can be
+
+```{r}
+library(sasr)
+
+df2sd(fev_data, "fev")
+
+code <- "
+ods output diffs = diff;
+PROC MIXED DATA = fev empirical cl method=reml;
+  CLASS RACE(ref = 'Asian') AVISIT(ref = 'VIS4') SEX(ref = 'Male') ARMCD(ref = 'PBO') USUBJID;
+  MODEL FEV1 = ARMCD / solution chisq;
+  REPEATED AVISIT / subject=USUBJID type=ar(1) r rcorr;
+  LSMEANS ARMCD / pdiff=all cl alpha=0.05 slice=AVISIT;
+RUN;"
+ret <- run_sas(code)
+df <- sd2df("diff")
+```
+
+In SAS result, the covariance matrix and degree of freedom are provided.
+The method for degree of freedom is "between within"
+
+```{r}
+cat(ret$LST)
+```
+
+# Implementations in `mmrm`
+
+Since using the "empirical" option, we are changing all standard errors and test statistics involving the
+fixed effect parameters, `beta_vcov`. So it can be similar to our other methods, "Satterthwaite" and "Kenward-Roger".
+So in `mmrm`, the method, stands for both "covariance adjust" and "degree of freedom calculation".
+In SAS, they are separated arguments.
```

Comment:
just to confirm: while in SAS they are separate, it is not possible to use empirical plus KR or ST in SAS?

Reply:
yes it is not possible; when using empirical option, the ddfm must be one of "between within", "contain" and "residule"; see 
![image](https://user-images.githubusercontent.com/36079400/211694495-ef28c3b7-d349-47f0-bb85-ac756008fc1a.png)

Reply:
cool thanks!